panos_upgrade_assurance/firewall_proxy.py

```diff
@@ -262,6 +262,11 @@ def is_panorama_connected(self) -> bool:
                 pan_connected = interpret_yes_no((pan_status_list[i].split(":")[1]).strip())
                 if pan_connected:
                     return True
+        elif pan_status_list_length in [3, 6]:
```

Comment:
2 comments:
* if the API provides actual XML output it can be much easier and consistent to parse this.
* Otherwise I would filter out the empty string elements from the list with the following statement between the lines 257 - 258 (after split and before taking len) - we can avoid the extra elif condition in this case.
`pan_status = list(filter(None, pan_status))`

Reply:
Unfortunately - it does not provide the structred XML output - it requires string parsing.
The idea with the filtered list though is great - I have refactored the code as such:

```
        pan_status_list = list(filter(None, pan_status.split("\n")))
        pan_status_list_length = len(pan_status_list)

        if pan_status_list_length in [3, 6]:
            for i in range(1, pan_status_list_length, 3):
                pan_connected = interpret_yes_no((pan_status_list[i].split(":")[1]).strip())
                if pan_connected:
                    return True
```

Now it should not matter if there is or is not a new empty line/s within the output.